Cargo.lock

```diff
@@ -1668,43 +1642,35 @@ dependencies = [
 [[package]]
 name = "ic-btc-types-internal"
 version = "0.1.0"
-source = "git+https://github.com/dfinity/ic?rev=d2564f5a929915b1f3588887d8b10004500f2ee2#d2564f5a929915b1f3588887d8b10004500f2ee2"
+source = "git+https://github.com/dfinity/ic?rev=1fc0208b9aeed0554b1be2711605e5b54ace9d6a#1fc0208b9aeed0554b1be2711605e5b54ace9d6a"
 dependencies = [
+ "candid",
+ "ic-btc-types",
  "ic-protobuf",
  "serde",
  "serde_bytes",
 ]
 
 [[package]]
 name = "ic-cdk"
-version = "0.3.3"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "606276ed1ce363eb9ccaf492e36fb40425417dcd4598f261d47e0ed6a1309faa"
-dependencies = [
- "candid",
- "cfg-if 1.0.0",
- "serde",
-]
-
-[[package]]
-name = "ic-cdk"
-version = "0.5.5"
+version = "0.6.4"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "141341c2cb26c4e5f196d5b64305c677b1abebe093eefbe2c622b4a2e6de8a9f"
+checksum = "d12a3ef5de121c699a4db746067d7810c68446804545dd9ecc39f05b6e4c5d74"
 dependencies = [
  "candid",
  "cfg-if 1.0.0",
+ "ic0",
```

Comment:
Looks suspicious. This might be why the linker is looking for the definition of things like canister_self_size.